scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -422,8 +422,20 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
     val pos = TokenStreamPosition(startTokenPos, endTokenPos + 1)
     result.withOrigin(Origin.Parsed(input, dialect, pos)).asInstanceOf[T]
   }
+
+  def atPosE[E, T <: Tree](start: Pos, end: Pos)(body: => Either[E, T]): Either[E, T] = {
+    val startTokenPos = start.startTokenPos
+    val result = body
+    var endTokenPos = end.endTokenPos
+    if (endTokenPos < startTokenPos) endTokenPos = startTokenPos - 1
+    val pos = TokenStreamPosition(startTokenPos, endTokenPos + 1)
+    result.map(_.withOrigin(Origin.Parsed(input, dialect, pos)).asInstanceOf[T])
```

Comment:
Will this work in Scala 2.11.12? I think Either is right biased starting from 2.12.0

Reply:
yeah test failed, will explicitly map right value